libs/common/src/main/java20/org/opensearch/common/round/RoundableFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.round;
+
+import org.opensearch.common.annotation.InternalApi;
+
+import jdk.incubator.vector.LongVector;
+import jdk.incubator.vector.VectorSpecies;
+
+/**
+ * Factory class to create and return the fastest implementation of {@link Roundable}.
+ *
+ * @opensearch.internal
+ */
+@InternalApi
+public final class RoundableFactory {
```

Comment:
I must be missing something, but there are two implementations of `org.opensearch.common.round.RoundableFactory`. This one is conditionally included if building with JDK>=20, but the other one is always included? How do we avoid JarHell with duplicate classes and/or avoid non-deterministic class loading?

Reply:
A multi-release JAR allows us to bundle original classes along with newer implementations targeting newer JDKs. At runtime, the JDK identifies the classes with the highest supported version. This process is deterministic.

Now there are two JDKs to consider:
1. The JDK used at build time – If using JDK >= 20, we set up Gradle to build a multi-release JAR with both sources. Otherwise, we create a regular JAR with the original classes only.
2. The JDK used at run time – If using JDK >= 20 and the JAR was built using JDK >= 20, then the overridden classes are picked up. Otherwise, the original implementation is used.